openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -781,149 +741,143 @@ def create_library_block(library_key, block_type, definition_id):
                     block_type=block_type, library_type=ref.type,
                 )
             )
-    lib_bundle = LibraryBundle(library_key, ref.bundle_uuid, draft_name=DRAFT_NAME)
-    # Total number of blocks should not exceed the maximum allowed
-    total_blocks = len(lib_bundle.get_top_level_usages())
-    if total_blocks + 1 > settings.MAX_BLOCKS_PER_CONTENT_LIBRARY:
+
+    # If adding a component would take us over our max, return an error.
+    component_count = publishing_api.get_all_drafts(ref.learning_package.id).count()
+    if component_count + 1 > settings.MAX_BLOCKS_PER_CONTENT_LIBRARY:
         raise BlockLimitReachedError(
-            _("Library cannot have more than {} XBlocks").format(settings.MAX_BLOCKS_PER_CONTENT_LIBRARY)
+            _("Library cannot have more than {} Components").format(
+                settings.MAX_BLOCKS_PER_CONTENT_LIBRARY
+            )
         )
+
     # Make sure the proposed ID will be valid:
-    validate_unicode_slug(definition_id)
+    validate_unicode_slug(block_id)
```

Comment:
@kdmccormick: reminder to check the implementation whether we'll have valid usage keys from this.

Reply:
@bradenmacdonald In addition to the usual suspects that `validate_slug/SlugField` checks for (letters, numbers, underscores, hyphens), `validate_unicode_slug` would allow for [modified letters](https://www.compart.com/en/unicode/category/Lm) and [other letters](https://www.compart.com/en/unicode/category/Lo). Even if OpaqueKeys tolerates modified letters and other letters, I'm not sure what all the downstream implications would be.

My sense is to limit it to classic ASCII SlugField for now, and loosen the restriction later if we feel confident. What do you think?

Reply:
Yes, OpaqueKeys and the content libraries API as it existed before this PR were designed to allow unicode slugs : https://github.com/openedx/opaque-keys/blob/f14bfd7b8970958e0d7f008a73902cb1f8f6686f/opaque_keys/edx/locator.py#L1542-L1543

My preference would be to just change one of the existing full tests like [this](https://github.com/openedx/edx-platform/blob/5ca009ca1bd61e4c433d6d8e51672f1c1ea6c005/openedx/core/djangoapps/content_libraries/tests/test_content_libraries.py#L211-L298) to use a unicode slug, which would basically ensure it works across the whole content libraries lifecycle (except probably wouldn't cover using it in a course). In just the same way as that test currently checks that unicode OLX works.

But I'm also fine with your suggested approach. If you want, you can make it ascii now and add this to the TODO list and I'll work on it in a separate PR with more tests.

Reply:
Cool, I was underestimating our ability to handle non-ascii here. I updated that exact test.